### PR TITLE
Update changelog and development model document

### PR DESCRIPTION
**Pull request by brycelelbach**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,10 +2,37 @@
...
+- #1082: Thrust now checks that it is compatible with the version of CUB found
+    in your include path, generating an error if it is not.
+  If you are using your own version of CUB, it may be too old.
+  It is recommended to simply delete your own version of CUB and use the
+    version of CUB that comes with Thrust.
+- #1089: C++03 and C++11 are deprecated.
+  Using these dialects will generate a compile-time warning.
+  These warnings can be suppressed by defining
+    `THRUST_IGNORE_DEPRECATED_CPP_DIALECT` (to suppress C++03 and C++11
+    deprecation warnings) or `THRUST_IGNORE_DEPRECATED_CPP11` (to suppress C++11
```
Comment (allisonvacanti) [inline CHANGELOG.md:25]: `THRUST_IGNORE_DEPRECATED_CPP_11`

File: DEVELOPMENT_MODEL.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
-# Thrust Branching and Development Model
+# Thrust Development Model
 
-The following is a description of how the Thrust development teams approaches branching and release tagging. This
-is a living document that will evolve as our process evolves.
+The following is a description of the basic development process that CUB follows. This is a living
```
Comment (allisonvacanti) [inline DEVELOPMENT_MODEL.md:3]: s/CUB/Thrust and CUB/

File: CHANGELOG.md (hunk 1 above)
Comment (brycelelbach) [reply CHANGELOG.md:25]: Fixed.

File: DEVELOPMENT_MODEL.md (hunk 2 above)
Comment (brycelelbach) [reply DEVELOPMENT_MODEL.md]: Fixed.

**Comment by allisonvacanti** [comment]
Rebased to update the submodule.

**Pull request merged**